resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -87,11 +87,15 @@ private[spark] object KubernetesClientUtils extends Logging {
    * Build a Config Map that will hold the content for environment variable SPARK_CONF_DIR
    * on remote pods.
    */
-  def buildConfigMap(configMapName: String, confFileMap: Map[String, String],
-      withLabels: Map[String, String] = Map()): ConfigMap = {
+  def buildConfigMap(
+     configMapName: String,
+     confFileMap: Map[String, String],
+     configMapNameSpace: String,
```

Comment:
Please try to get namespace via `confFileMap` instead changing this method signature.

Reply:
Done, buildConfigMap method could get namespace via `confFileMap`